plugins/uncnd_branch_print.c

```diff
@@ -0,0 +1,71 @@
+/*
+  This file is part of MAMBO, a low-overhead dynamic binary modification tool:
+      https://github.com/beehive-lab/mambo
+
+  Copyright 2021 The University of Manchester
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifdef PLUGINS_NEW
+
+#include <stdio.h>
+#include <assert.h>
+#include <locale.h>
+#include <inttypes.h>
+#include "../plugins.h"
+
+int uncnd_branch_print_pre_thread_handler(mambo_context *ctx) {
+  uint64_t *uncnd_br_count = mambo_alloc(ctx, sizeof(uint64_t));
+  *uncnd_br_count = 0;
+  assert(uncnd_br_count != NULL);
+  mambo_set_thread_plugin_data(ctx, uncnd_br_count);
+  return 0;
+}
+
+// Called when a thread exits, or in all threads when the process exits
+int uncnd_branch_print_post_thread_handler(mambo_context *ctx) {
+  uint64_t *uncnd_br_count = mambo_get_thread_plugin_data(ctx);
+  fprintf(stderr, "%'lu unconditional branches executed in thread %d\n", *uncnd_br_count, mambo_get_thread_id(ctx));
+  mambo_free(ctx, uncnd_br_count);
+  return 0;
+}
```

Comment:
`uncnd_branch_print_post_thread_handler` is not incrementing a global counter.
Once a treads finishes you loose the number of unconditional branches that were executed in the thread.